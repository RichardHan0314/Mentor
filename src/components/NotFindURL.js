import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import BackToHomeBtn from './BackToHomeBtn';

const NotFindURL = () => {

    return (
        <Wrapper>
            <Content>Page Not Found</Content>
            <Image src="https://blog.vverh.digital/wp-content/uploads/2020/06/oblojka-404.png" alt="404image" />
            <Link to='/'>
                <BackToHomeBtn />
            </Link>
        </Wrapper>
    );
};

const Wrapper = styled.div`
    padding: 60px 0;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    height: 100%;
    margin: 0;
    background: #121212;
`;

const Image = styled.img`
    width: 60%;
    height: auto;
`;

const Content = styled.p`
    color: #fff;
    font-size: 60px;
    font-family: sans-serif;
    text-align: center;
    margin: 0;
`;

export default NotFindURL;
