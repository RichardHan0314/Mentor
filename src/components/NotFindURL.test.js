import React from 'react';
import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import { MemoryRouter, Link } from 'react-router-dom';
import NotFindURL from './NotFindURL';

describe('NotFindURL component', () => {
    it('Render NotFindURL snapshot', () => {
        const component = shallow(
            <MemoryRouter>
                <NotFindURL />
            </MemoryRouter>
        );
        expect(component).toMatchSnapshot();
    });

    it('should includes link to home path', () => {
        const component = shallow(
            <MemoryRouter>
                <NotFindURL>
                    <Link to="/" />
                </NotFindURL>
            </MemoryRouter>
        );

        // screen.debug();
        expect(component.find(Link).props().to).toBe('/');
    });
});