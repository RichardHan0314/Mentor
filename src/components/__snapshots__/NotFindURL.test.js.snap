// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFindURL component Render NotFindURL snapshot 1`] = `ShallowWrapper {}`;
